/buy test cases creation

annaw3558 opened the pull request:
Added all the frontend test cases for R6.
I made a new python file to keep it organized. Some code had to be changed in the frontend.py and index.html to fix errors that were showing up when running the test cases. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

annaw3558 committed:
literally everything

annaw3558 committed:
final little changes

deleted some objects that weren't needed for mocking

annaw3558 committed:
<3

michael-fourie reviewed (approved):
Good job creating the test cases, and cleaning up the frotnend code. You introduce no new bugs which is good, so I will approve this request

annaw3558 committed:
final test case

added in test case for ticket name does not exist in the database

Slabrosse reviewed (approved):
This all looks good, thank you for solving all the bugs that we were having with the submit button and some of the POST requests. Used the changes in my code and work perfectly.

annaw3558 committed:
bug fix

added in when the user actually buys the ticket, the cost is decremented from their account balance

GabrielleGravel reviewed (approved):
Everything looks good, glad we could resolve those issues. test cases for buy fit requirements and merge well. awesome job 💯

michael-fourie committed:
Merge branch 'master' into buy_testcases
